share/spice/today_in_history/today_in_history.js

```diff
@@ -1,33 +1,158 @@
-function ddg_spice_today_in_history(api_response) {
+(function (env) {
     "use strict";
+    env.ddg_spice_today_in_history = function(api_response) {
 
-    if (!api_response) {
-        return Spice.failed('today_in_history');
-    }
+        if (!api_response) {
+            return Spice.failed('today_in_history');
+        }
+    
+        // Extract our query	
+        var script = $('[src*="/js/spice/today_in_history/"]')[0],
+        source = $(script).attr('src'),
+        ourquery = source.match(/today_in_history\/([^\/]+)/)[1];
+        ourquery = ourquery.replace("_", " ");    
+    
+        // Extract data
+        var item
+        for (var key in api_response.query.pages) {
+            if (api_response.query.pages[key].hasOwnProperty("revisions"))
+                item = api_response.query.pages[key].revisions[0]["*"];
+        }
 
-    var item = $( $.parseXML(api_response) ).find('item');
+        if(!item) {
+            return Spice.failed('today_in_history');
+        }
 
-    if (!item) {
-        return Spice.failed('today_in_history');
-    }
+        // Extract the data we want to organize    	
+        var temp_a = item.split("==Events==");
+        var temp_b = temp_a[1].split("==Births==");
+        var temp_c = temp_b[1].split("==Deaths==");
+        var temp_d = temp_c[1].split("==Holidays and observances==");
+        
+        var events = $.trim(temp_b[0]).split("\n");
+      
+        // We will use these when we get the filters
+        // var births = $.trim(temp_c[0]).replace(/\<\!--(.|\n)*/,"").split("\n");
+        // var deaths = $.trim(temp_d[0]).replace(/\<\!--(.|\n)*/,"").split("\n");
+    
 
-    var title = item.find('title').text();
-    var link  = item.find('link').text();
-    var text  = item.find('description').text();
+        // Organize data in array
+        var temp;
+        var array = [];
+        var temp_string;
+        var toggle = 0;
+        for (var i = 0; i < events.length; i++) {
+            temp = events[i].split(" &ndash; ");
+            temp_string = (temp.slice(1)).join(" ");
+            if (temp_string.length > 0) {
+                array.push( {year: temp[0].replace(/\*/,""), str: temp_string, ttype: "event"});
+            }
+        }
+        
+        Spice.add({
+            id: "today_in_history",
+            name: "Today in History",
+            data: array,
+            meta: {
+                itemType: "historical events for " + ourquery,
+                sourceUrl: 'http://en.wikipedia.org/wiki/'+ ourquery,
+                sourceName: 'Wikipedia',
+                minItemsForModeSwitch: '9999',
```

Comment:
There's a nicer way to do this now. You can either disable it with this:
```
disableModeSwitch: true
```
or you can hide it entirely with this:
```
hideModeSwitch: true
```